staging/src/k8s.io/apiserver/pkg/endpoints/request/requestinfo.go

```diff
@@ -178,7 +178,12 @@ func (r *RequestInfoFactory) NewRequestInfo(req *http.Request) (*RequestInfo, er
 
 	// URL forms: /namespaces/{namespace}/{kind}/*, where parts are adjusted to be relative to kind
 	if currentParts[0] == "namespaces" {
-		if len(currentParts) > 1 {
+		// When creating a new namespace using server-side apply a PATCH request is
+		// sent at /namespaces/new-namespace.  In this case, we don't want to add
+		// the namespace to the request.
+		if len(currentParts) == 2 && requestInfo.Verb == "patch" {
+			requestInfo.Namespace = metav1.NamespaceNone
```

Comment:
this is not the right layer to make this change at... we should understand where the data is being set into the object and why it differs for patch requests

Reply:
for request-info purposes, updates to a namespace *are* namespaced... setting this to "" here means that a user that was previously authorized to patch a specifically named namespace will get rejected

Reply:
Oh, makes sense! thanks for spotting it!